scripts/action-handler.js

```diff
@@ -1168,14 +1169,9 @@ Hooks.once('tokenActionHudCoreApiReady', async (coreModule) => {
             const encodedValue = [actionType, id].join(this.delimiter)
             const img = coreModule.api.Utils.getImage(entity)
             const icon1 = this.#getActivationTypeIcon(entity?.system?.activation?.type)
-            let icon2 = null
-            let info = null
-            if (entity.type === 'spell') {
-                icon2 = this.#getPreparedIcon(entity)
-                if (this.displaySpellInfo) info = this.#getSpellInfo(entity)
-            } else {
-                info = this.#getItemInfo(entity)
-            }
+            const icon2 = entity.type === 'spell' && this.showUnpreparedSpells ? this.#getPreparedIcon(entity) : null
+            const icon3 = entity.type === 'spell' && this.displaySpellInfo ? this.#getConcentrationIcon(entity) : null
```

Comment:
Can we move the check for `entity.type` and `this.showUnpreparedSpells` to the `getPreparedIcon` method and likewise for the `getConcentrationIcon`. That'll keep this nice and tidy.

Reply:
Check, will do!